packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -992,5 +1014,29 @@ class _RepeatingSimulation extends Simulation {
   double dx(double timeInSeconds) => (max - min) / _periodInSeconds;
 
   @override
-  bool isDone(double timeInSeconds) => false;
+  bool isDone(double timeInSeconds) {
+    if (repeatTimes == null) {
+      // if repeat times is null it is to be consider as repeated simulation
+      return false;
+    }
+
+    // total simulation elapsed time [Fixed to 3 decimal points]
+    final double totalTimeInSeconds = double.parse(
+      (timeInSeconds + _initialT).toStringAsFixed(3),
+    );
+
+    // if reverse simulation is [true], then we shall double the repeat times
+    // else, use the given value
+    final int effectiveRepeatTimes = repeatTimes! * (reverse ? 2 : 1);
+
+    // [exitTimeInSeconds] is the time in seconds which will help
+    // to determine when to complete the simulation. [Fixed to 3 decimal points]
+    final double exitTimeInSeconds = double.parse(
+      (effectiveRepeatTimes * _periodInSeconds).toStringAsFixed(3),
+    );
```

Comment:
Instead of computing these during each function call, could we set them as `late final` class members or something similar?

Also, is there a reason why we're rounding to 3 decimal places here?

Reply:
Added private variable to calculate exit time in seconds only once for entire simulation.

Reason behind rounding it to 3 decimal points is, usually we don't need 7-8 decimal points value as its already in seconds so 3 decimal points shall be enough. Also, depending upon the device performance and all - simulation might differ if we compare whole decimal value. Please suggest your thoughts.